src/plugins/dashboard/server/saved_objects/dashboard_migrations.ts

```diff
@@ -111,7 +111,7 @@ export const dashboardSavedObjectTypeMigrations = {
    * in that version. So we apply this twice, once with 6.7.2 and once with 7.0.1 while the backport to 6.7
    * only contained the 6.7.2 migration and not the 7.0.1 migration.
    */
-  '6.7.2': flow<SavedObjectMigrationFn>(migrateMatchAllQuery),
-  '7.0.0': flow<(doc: SavedObjectUnsanitizedDoc) => DashboardDoc700To720>(migrations700),
-  '7.3.0': flow<SavedObjectMigrationFn>(migrations730),
+  '6.7.2': flow<SavedObjectMigrationFn<any, any>>(migrateMatchAllQuery),
+  '7.0.0': flow<SavedObjectMigrationFn<any, DashboardDoc700To720['attributes']>>(migrations700),
+  '7.3.0': flow<SavedObjectMigrationFn<any, any>>(migrations730),
```

Comment:
```suggestion
  '6.7.2': flow<typeof migrateMatchAllQuery>(migrateMatchAllQuery),
  '7.0.0': flow<typeof migrations700>(migrations700),
  '7.3.0': flow<SavedObjectMigrationFn<any, any>>(migrations730),
```

We can't follow the same pattern with `migrations730` because it's not typed as a `SavedObjectMigrationFn`.